### PR TITLE
Remove unneeded path parameter for creating an issue tracker from apidocs documentation

### PR DESCRIPTION
**Pull request by eduardoj**
The `POST` route for issue trackers doesn't require an issue tracker id in path parameters. And the `create` method in the controller doesn't make use of any path parameters.

This is how issue_trackers routes looks like now:

```
issue_tracker_issue GET    /issue_trackers/:issue_tracker_id/issues/:id(.:format) issues#show                                                            
     issue_trackers GET    /issue_trackers(.:format)                              issue_trackers#index                                                   
                    POST   /issue_trackers(.:format)                              issue_trackers#create                                                  
      issue_tracker GET    /issue_trackers/:id(.:format)                          issue_trackers#show                                                    
                    PATCH  /issue_trackers/:id(.:format)                          issue_trackers#update                                                  
                    PUT    /issue_trackers/:id(.:format)                          issue_trackers#update                                                  
                    DELETE /issue_trackers/:id(.:format)                          issue_trackers#destroy                                                 
```

Also replace `bnc` as an example of ids with `<id>`, in the comments, for the description of the route related to the controller method.

-----------------

If this PR requires any particular action or consideration before deployment,
please check the reasons or add your own to the list:

* [ ] Requires a database migration that can cause downtime. Postpone deployment until maintenance window[1].
* [ ] Contains a data migration that can cause temporary inconsistency, so should be run at a specific point of time.
* [ ] Changes some configuration files (e.g. options.yml), so the changes have to be applied manually in the reference server.
* [ ] A new Feature Toggle[2] should be enabled in the reference server.
* [ ] Proper documentation or announcement has to be published upfront since the introduced changes can confuse the users.

[1] https://github.com/openSUSE/open-build-service/wiki/Deployment-of-build.opensuse.org#when-there-are-migrations
[2] https://github.com/openSUSE/open-build-service/wiki/Feature-Toggles-%28Flipper%29#you-want-real-people-to-test-your-feature

### CODE REVIEW

File: docs/api/api/api.txt [hunk 1]
```diff
@@ -47,6 +47,12 @@ XmlResult: issue_trackers
 
 <name>: Issue tracker name
 
+POST /issue_trackers
+
+  Create new issue tracker.
```
Comment (dmarcoux) [inline docs/api/api/api.txt:52]: Creating something already implies that it's new. I suggest going with:

```suggestion
  Create an issue tracker.
```

File: src/api/app/controllers/issue_trackers_controller.rb [hunk 2]
```diff
@@ -46,7 +46,7 @@ def create
     end
   end
 
-  # PUT /issue_trackers/bnc
+  # PUT /issue_trackers/<id>
```
Comment (dmarcoux) [inline src/api/app/controllers/issue_trackers_controller.rb:49]: This is a name, not an id. Same for the other change in this file.

```suggestion
  # PUT /issue_trackers/<name>
```

**Review by dmarcoux — CHANGES_REQUESTED** [review]
The routes are expecting an id because this is the default for resources in Rails. However, it is really a name. We are retrieving the issue trackers with _IssueTracker.**find_by_name**(params[:id])_, so it's clearly a name.

I suggest renaming the default param from `id` to `name` since this is what it really is. This involves changing the routes. We would need to add `param: :name` to the resources defined in https://github.com/openSUSE/open-build-service/blob/6bf4751cebcd518bbeaaafc98b5113c0ec230842/src/api/config/routes/api_routes.rb#L74

We can then clear out the inconsistencies in the code and only use `param[:name]`. In the end, I wonder if those changes are worth the time. I understand if you don't want to go this far.

**Comment by dmarcoux** [comment]
Instead of `not needed` in the commit message, I suggest using `unneeded`

**Comment by eduardoj** [comment]
@dmarcoux, I applied some of your sugestions in the existing commit, and added a couple of commits more:
- also following your suggestion, to use only `param[:name]`
- the last commit, to remove a line that was not used.

> We can then clear out the inconsistencies in the code and only use `param[:name]`. In the end, I wonder if those changes are worth the time. I understand if you don't want to go this far.

Let's do it, and hope for the best! :facepalm:

Please have a look again.

**Review by danidoni — APPROVED** [review]

**Review by dmarcoux — APPROVED** [review]

**Pull request merged**